utils/log_parser.py

```diff
@@ -15,136 +15,214 @@
-        if node not in self.uniqueVertexMap:
-            self.uniqueVertexMap[node] = self.uniqueVertexNo
-            self.uniqueVertexNo += 1
-        return "v" + str(self.uniqueVertexMap[node])
-
-    def dump_label(self, node):
-        labelStr = "{"
-        labelStr += " {<Inputs>Inputs}|"
-        labelStr += "{" + node.get_kind_name() + "\lname: " + \
-            node.get_name() + "}|"
-        labelStr += " {<Outputs>Outputs}"
-        labelStr += "}"
+        self.vertices_: List[str] = []
+        self.edges_: List[str] = []
+        self.uniqueVertexMap_: Dict[Node, int] = {}
+        self.uniqueVertexNo_: int = 0
+        self.colors_: List[str] = [
+            "AliceBlue",
+            "CadetBlue1",
+            "Coral",
+            "DarkOliveGreen1",
+            "DarkSeaGreen1",
+            "GhostWhite",
+            "Khaki1",
+            "LavenderBlush1",
+            "LemonChiffon1",
+            "LightSkyBlue",
+            "MistyRose1",
+            "MistyRose2",
+            "PaleTurquoise2",
+            "PeachPuff1",
+            "PowderBlue",
+            "Salmon",
+            "Thistle1",
+            "Thistle3",
+            "Wheat1",
+            "Yellow2"]
+
+    def get_unique_vertex_name(self, node: Node) -> str:
+        """Get the unique vertex name given a Node object. """
+
+        if node not in self.uniqueVertexMap_:
+            self.uniqueVertexMap_[node] = self.uniqueVertexNo_
+            self.uniqueVertexNo_ += 1
+        return "v" + str(self.uniqueVertexMap_[node])
+
+    def dump_label(self, node: Node) -> str:
+        """Returns the string for the label of the given node. """
+
+        labelStr = "{ {<Inputs>Inputs}|{"
+        labelStr += node.get_kind_name() + "\lname: " + \
+            node.get_name()
+        labelStr += "}| {<Outputs>Outputs}}"
         return labelStr
 
-    def get_color(self, node):
-        idx = hash(node.get_kind_name()) % len(self.colors)
-        return self.colors[idx]
+    def get_color(self, node: Node) -> str:
+        """Returns the color for the given node. """
+
+        idx = hash(node.get_kind_name()) % len(self.colors_)
+        return self.colors_[idx]
+
+    def dump_node(self, node: Node) -> None:
+        """Generates the dotty information for the given node. """
 
-    def dump_node(self, node):
         if not node:
             return
-        nodeStr = self.get_unique_vertex_name(node) + "[\n"
-        nodeStr += "\tlabel = \""
-        nodeStr += self.dump_label(node)
-        nodeStr += "\"\n"
+        nodeStr = f"{self.get_unique_vertex_name(node)}[\n"
+        nodeStr += f"\tlabel = \"{self.dump_label(node)}\"\n"
         nodeStr += "\tshape = \"record\"\n"
         nodeStr += "\tstyle=\"filled,rounded\"\n"
-        nodeStr += "\tfillcolor="
-        nodeStr += self.get_color(node)
-        nodeStr += "\npenwidth = 2];\n"
+        nodeStr += f"\tfillcolor={self.get_color(node)}\n"
+        nodeStr += "penwidth = 2];\n"
```

Comment:
You should be able to combine these lines into into assignment with a multiline f-string, e.g.:
```
nodeStr = f"""{self.get_unique_vertex_name(node)}[\n
              \tlabel = \"{self.dump_label(node)}\"\n
              \tshape = \"record\"\n
               ...
               penwidth = 2];\n"""
```